src/pages/overview/document-generation-api/templatetags.md

```diff
@@ -245,6 +245,32 @@ Default extension is vertical. If cell-extension construct is not provided, then
 
 ![Table gets horizontally extended as table columns are repeated](../images/horizontal-extension1.png)
 
+### Add index based conditions
+Conditionals can also be added inside tables. Condition evaluation is similar to that outside table, but we provide an additional construct
```

Comment:
Conditional Sections are now supported inside Table and List. Use of '_expr-context_' construct is **mandatory** to define conditionals inside tables and lists. 'expr-context' 
Below is the example showing the usage of 'expr-context' and conditionals inside tables and lists.

Reply:
Use of expr-context is not mandatory for conditions inside tables as one can also use normal conditions based on  fields present outside the array of objects.

Reply:
got it.

Reply:
I raised a comment on this before, but do we need to use the expr-context expression? I don't understand why that is necessary when inside a table, I can already refer to a value that's in the array. So why do we need to provide additional context for a condition?

Reply:
@Dheeraj3reddy Can you add in your thought here.

Reply:
Consider the expression expr(company.supportsX=”TRUE”). This is a condition which needs to be evaluated separately for each company i.e., each row in the final output has a different output based on its supportsX field. As per jsonata, the expression company.supportsX=”TRUE” cannot be evaluated directly as left-hand side of equation is an array of objects [“TRUE”,”FALSE”,”TRUE”] and right-hand side is a normal string. This is different from normal text tags being used inside table (like company.name as in those we don’t need any comparison as such, we just get an array result from jsonata [“ABC”,”DEF”,”XYZ”] and populate the table based on that. The key reason for conditions requiring extra context is that jsonata does not support comparison (operators like =, !=, >, <, …) between an array of objects and normal string.